azure_edge_iot_ops_jumpstart/aio_esa/artifacts/LogonScript.ps1

```diff
@@ -0,0 +1,444 @@
+Write-Host "Open Service Mesh (OSM) installed successfully."
+# Disable ACStor for single-node cluster
+Write-Host "Disabling ACStor for single-node cluster..."
+# Create the config.json file
+$acstorConfig = @{
+    "hydra.acstorController.enabled" = $false
+    "hydra.highAvailability.disk.storageClass" = "local-path"
+ }
+ $acstorConfigJson = $acstorConfig | ConvertTo-Json -Depth 100
+ Set-Content -Path "config.json" -Value $acstorConfigJson
+Write-Host "ACStor disabled for single-node cluster."
+Write-Host "Checking if Edge Storage Accelerator Arc Extension is installed..."
+$extensionExists = az k8s-extension show --resource-group "$env:resourceGroup" --cluster-name "$env:arcClusterName" --cluster-type connectedClusters --name hydraext --query "extensionType" --output tsv
+if ($extensionExists -eq "microsoft.edgestorageaccelerator") {
+   Write-Host "Edge Storage Accelerator Arc Extension is already installed."
+} else {
+   Write-Host "Installing Edge Storage Accelerator Arc Extension..."
+   az k8s-extension create --resource-group "$env:resourceGroup" --cluster-name "$env:arcClusterName" --cluster-type connectedClusters --name hydraext --extension-type microsoft.edgestorageaccelerator --config-file "config.json"
+   Write-Host "Edge Storage Accelerator Arc Extension installed successfully."
+}
+
+# Create Kubernetes secret for Azure Storage account
+Write-Host "Creating Kubernetes secret for Azure Storage account..."
+$secretName = "$env:storageAccountName-secret"
+Add-AzureStorageAccountSecret -ResourceGroup $env:resourceGroup -StorageAccount $env:storageAccountName -Namespace "default" -SecretName "esa-secret"
+Write-Host "Kubernetes secret created successfully."
+Write-Host "Downloading pv.yaml file..."
+$pvYamlUrl = "https://raw.githubusercontent.com/fcabrera23/azure_arc/scenarios-esa/azure_edge_iot_ops_jumpstart/aio_esa/yaml/pv.yaml"
+$pvYamlPath = "pv.yaml"
+Invoke-WebRequest -Uri $pvYamlUrl -OutFile $pvYamlPath
+# Update the secret name and container name in the pv.yaml file
+#$pvYamlContent = Get-Content -Path $pvYamlPath -Raw
+#$pvYamlContent = $pvYamlContent -replace '\${CONTAINER_NAME}-secret', $secretName
+#$pvYamlContent = $pvYamlContent -replace '\${CONTAINER_NAME}', $env:storageContainer
+#Set-Content -Path $pvYamlPath -Value $pvYamlContent
+# Apply the pv.yaml file using kubectl
+Write-Host "Applying pv.yaml configuration..."
+kubectl apply -f $pvYamlPath
+Write-Host "pv.yaml configuration applied successfully."
+Write-Host "Downloading esa-deploy.yaml file..."
+$esadeployYamlUrl = "https://raw.githubusercontent.com/fcabrera23/azure_arc/scenarios-esa/azure_edge_iot_ops_jumpstart/aio_esa/yaml/esa-deploy.yaml"
+$esadeployYamlPath = "esa-deploy.yaml"
+Invoke-WebRequest -Uri $esadeployYamlUrl -OutFile $esadeployYamlPath
+# Apply the p-deploy.yaml file using kubectl
+Write-Host "Applying esadeploy.yaml configuration..."
+kubectl apply -f $esadeployYamlPath
+Write-Host "esa-deploy.yaml configuration applied successfully."
```

Comment:
any environment cleanup needed here?

What about Stop-Transcript?

Reply:
Its already in there

# Stop the PowerShell process monitoring Kubernetes pods

Stop-Process -Id $kubectlMonShell.Id

# Remove temporary files and directories

Remove-Item -Path "$installDir\$zipFile" -Force -ErrorAction SilentlyContinue
Remove-Item -Path "$installDir\AKS-Edge-main" -Recurse -Force -ErrorAction SilentlyContinue
Remove-Item -Path "AzureConnectedMachineAgent.msi" -Force -ErrorAction SilentlyContinue

# Stop the transcript

Stop-Transcript